src/tests/ftest/osa/osa_offline_extend.py

```diff
@@ -0,0 +1,196 @@
+        n = random.randint(1, 7)
+        target_list.append(n)
+        target_list.append(n+1)
+
+        # Extend a rank (or server)
+        # rank index starts from zero
+        rank = total_servers
+
+        for val in range(0, num_pool):
+            pool[val] = TestPool(self.context,
+                                 dmg_command=self.get_dmg_command())
+            pool[val].get_params(self)
+            # Split total SCM and NVME size for creating multiple pools.
+            pool[val].scm_size.value = int(pool[val].scm_size.value /
+                                           num_pool)
+            pool[val].nvme_size.value = int(pool[val].nvme_size.value /
+                                            num_pool)
+            pool[val].create()
+            pool_uuid.append(pool[val].uuid)
+            self.pool = pool[val]
+            if data:
+                self.write_single_object()
+        # Start the additional servers and extend the pool
+        self.log.info("Extra Servers = %s", self.extra_servers)
+        self.start_additional_servers(self.extra_servers)
+        # Give sometime for the additional server to come up.
+        time.sleep(5)
+
+        # Extend the pool_uuid, rank and targets
+        for val in range(0, num_pool):
+            self.pool = pool[val]
+            scm_size = self.pool.scm_size
+            nvme_size = self.pool.nvme_size
+            self.pool.display_pool_daos_space("Pool space: Beginning")
+            pver_begin = self.get_pool_version()
+            self.log.info("Pool Version at the beginning %s", pver_begin)
+            output = self.dmg_command.pool_extend(self.pool.uuid,
+                                                  rank, scm_size,
+                                                  nvme_size)
+            self.log.info(output)
+
+            pver_extend = self.get_pool_version()
+            fail_count = 0
+            while fail_count <= 20:
+                pver_extend = self.get_pool_version()
+                time.sleep(15)
+                fail_count += 1
+                if pver_extend > pver_begin:
+                    break
+
+            pver_extend = self.get_pool_version()
+            self.log.info("Pool Version after extend %d", pver_extend)
+            # Check pool version incremented after pool extend
+            self.assertTrue(pver_extend > pver_begin,
+                            "Pool Version Error:  After extend")
+
+        for val in range(0, num_pool):
+            display_string = "Pool{} space at the End".format(val)
+            pool[val].display_pool_daos_space(display_string)
+            pool[val].destroy()
+
+    def test_osa_offline_extend(self):
+        """
+        JIRA ID: DAOS-4751
+
+        Test Description: Validate Offline Extend
+
+        :avocado: tags=all,pr,hw,large,osa,osa_extend,offline_extend
+        """
+        # Perform extend testing with 1 pool
+        self.run_offline_extend_test(1)
```

Comment:
I'm curious. Why do you run the test with 1 pool only? Since you have the infrastructure, wouldn't it make sense to test with different number of pools?

Reply:
Initially, I wanted to test with one pool. Once the test passes, I can increase the testing with multiple pools (based on overall time taken by the test).

Reply:
Once we address DAOS-5869, we can work on increasing the coverage with multiple pools. I have created a ticket DAOS-5870 to enhance the test to handle multiple pools.

Reply:
I see. Thanks.